plugins/org.eclipse.fordiac.ide.library/src/org/eclipse/fordiac/ide/library/DownloadResult.java

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Patrick Aigner - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.library;
+
+// result is provided with status OK - it is null in all other cases
+public record DownloadResult<T>(T result, Status status, String message) {
+
+	public DownloadResult(final T result) {
+		this(result, Status.OK, null);
```

Comment:
To ensure "graceful" failure you could check if result is `null` here and create an object that indicates an error
```suggestion
		this(result, result != null ? Status.OK : Status.ERROR, null);
```

Reply:
added this and a "Null result" for the message